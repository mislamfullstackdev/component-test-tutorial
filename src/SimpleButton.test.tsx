import {render, screen} from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { SimpleButton } from "./SimpleButton"

test("Click the button to switch ON/OFF display", async () => {
    render(<SimpleButton/>);
    const simpleButton = screen.getByRole("button");
    expect(simpleButton).toHaveTextContent(("OFF"));
    await userEvent.click(simpleButton);
    expect(simpleButton).toHaveTextContent("ON");
});

// Added Snapshot testing
test("Immediately after being drawn, it is displayed as OFF", () => {
    const view = render(<SimpleButton/>);
    expect(view.container).toMatchSnapshot();
});