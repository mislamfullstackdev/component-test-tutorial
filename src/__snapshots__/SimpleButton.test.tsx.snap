// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Immediately after being drawn, it is displayed as OFF 1`] = `
<div>
  <button>
    OFF
  </button>
</div>
`;
